src/tslint.ts

```diff
@@ -91,6 +91,10 @@ class Linter {
             sourceFile = getSourceFile(this.fileName, this.source);
         }
 
+        if (sourceFile === undefined) {
+            throw new Error("Invalid source code: " + this.fileName);
```

Comment:
I'm not sure that tsc only checks file extension.  `sourceFile` may be `undefined` because of other than file extension.